### PR TITLE
CONSOLE-3685: Convert components in RBAC folder from class component …

### PR DESCRIPTION
**Pull request by tvu20**
…to functional component

The following files in frontend/public/components/RBAC have been converted from a class component to a functional component in order to use react-router v6 hooks:

edit-rule.jsx
bindings.jsxs

### CODE REVIEW

**Comment by tvu20** [comment]
This is technical debt cleanup and has no user-facing impact, so I am manually assigning docs and px approvals.

/assign @yapei for QE approval.
/label docs-approved
/label px-approved

File: frontend/public/components/RBAC/bindings.jsx [hunk 1]
```diff
@@ -430,21 +430,30 @@ const Section = ({ label, children }) => (
...
-    const existingData = _.pick(props.obj, [
-      'metadata.name',
-      'metadata.namespace',
-      'roleRef',
-      'subjects',
-    ]);
-    existingData.kind = props.kind;
-    const { subjectKind, subjectName } = this.props.fixed.subjectRef || {};
-    const data = _.defaultsDeep({}, props.fixed, existingData, {
+const BaseEditRoleBindingWithTranslation = (props) => {
```
Comment (rhamilto) [inline frontend/public/components/RBAC/bindings.jsx:433]: `WithTranslation` is no longer needed now that this is a function component and you are using the hook below.  Can rename to `BaseEditRoleBinding` and remove `const BaseEditRoleBinding = withTranslation()(BaseEditRoleBindingWithTranslation);` on line 745.

**Review by rhamilto — COMMENTED** [review]
Nice work, @tvu20.  One comment.

**Comment by jhadvig** [comment]
>Wondering if we want to take the opportunity to delete since these have been disabled for a very long time.

@rhamilto I would go ahead and remove the code if its not used. If we ever need to revisit it, GitHub will keep the history 🏰

**Comment by rhamilto** [comment]
@tvu20, please go ahead and remove `EditRulePage` since it is not in use.  Sorry to throw away the work you did.

**Comment by XiyunZhao** [comment]
This PR has been tested in private build, currently no issue was found
/label qe-approved

**Comment by rhamilto** [comment]
@tvu20, please also remove the commented out routes in  See https://github.com/openshift/console/blob/24559efd18f032a66b6baad58d40715c1d36c0fa/frontend/public/components/app-contents.tsx#L357-L366.

**Pull request merged**